app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -204,15 +206,25 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
             requireContext(),
             requireContext().settings(),
             interactor,
-            historyStorageProvider(),
             isPrivate,
             binding.toolbar,
-            requireComponents.core.engine,
             fromHomeFragment
         ).also {
             inlineAutocompleteEditText = it.view.findViewById(R.id.mozac_browser_toolbar_edit_url_view)
         }
 
+        val engineForSpeculativeConnects = if (!isPrivate) requireComponents.core.engine else null
+        if (requireContext().settings().shouldAutocompleteInAwesomebar) {
+            ToolbarAutocompleteFeature(
+                binding.toolbar,
+                engineForSpeculativeConnects,
+                { store.state.searchEngineSource.searchEngine?.type != SearchEngine.Type.APPLICATION }
```

Comment:
let's add the named params to make it easier to understand

Reply:
Agreed.  Thanks